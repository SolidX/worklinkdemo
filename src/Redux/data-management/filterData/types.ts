import { IWorkOrder } from '../../../data';
import { string } from 'prop-types';

// Constants
export enum ActionTypes {
  CHANGE_FILTER_BY_STATUS = '@@filterData/CHANGE_FILTER_BY_STATUS',
  SORT_BY = '@@filterData/SORT_BY'
}

export interface IStatus {
  id: 'completed' | 'unassigned' | 'on hold' | 'assigned';
  inuse: boolean;
}

export interface ISortBy {
  sortBy: 'asc_date' | 'desc_date';
}
export interface IFilterByStatusProps {
  filters: IStatus[];
  // filteredWorkOrders: IWorkOrder[];
  sortBy: ISortBy;
}
export interface IChangeFilterByStatusAction {
  type: ActionTypes.CHANGE_FILTER_BY_STATUS;
  payload: { statusId: string };
}

export interface ISortByAction {
  type: ActionTypes.SORT_BY;
  payload: { sortBy: ISortBy };
}
