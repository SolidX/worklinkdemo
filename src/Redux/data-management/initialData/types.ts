import { IWorkOrder, IWorker } from '../../../data';

// Constants
export enum ActionTypes {
  GET_INITIAL_DATA_BEGIN = '@@initialData/GET_INITIAL_DATA_BEGIN',
  GET_INITIAL_DATA_SUCCESS = '@@initialData/GET_INITIAL_DATA_SUCCESS',
  GET_INITIAL_DATA_FAILURE = '@@initialData/GET_INITIAL_DATA_FAILURE',
  CHANGE_FILTER = '@@initialData/CHANGE_FILTER'
}

export interface IGetInitialDataBeginAction {
  type: ActionTypes.GET_INITIAL_DATA_BEGIN;
}

export interface IGetInitialDataSuccessAction {
  type: ActionTypes.GET_INITIAL_DATA_SUCCESS;
  payload: { initialData: IInitialDataProps };
}

export interface IGetInitialDataFailureAction {
  type: ActionTypes.GET_INITIAL_DATA_FAILURE;
  payload: { error: any };
}

export interface IInitialDataProps {
  workOrders: IWorkOrder[];
  employees: IWorker[];
  loading: boolean;
  filters: IFilter[];
  error: any;
}

export interface IInitialApiData {
  workOrders: IWorkOrder[];
  employees: IWorker[];
}

export interface IFilter {
  id: 'completed' | 'unassigned' | 'on hold';
  inuse: boolean;
}

export interface IChangeFilterAction {
  type: ActionTypes.CHANGE_FILTER;
  payload: { filter: IFilter };
}
