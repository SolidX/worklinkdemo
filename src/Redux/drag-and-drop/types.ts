// TeamMember Object
export interface ITeamMember {
  id: string;
}

// Constants
export const enum ActionTypes {
  TOGGLE_TEAM_MEMBER = '@@drag-and-drop/TOGGLE_TEAM_MEMBER',
  // REMOVE_TEAM_MEMBER = '@@drag-and-drop/REMOVE_TEAM_MEMBER',
  ASSIGN_TEAM = '@@drag-and-drop/ASSIGN_TEAM'
}

export interface IToggleTeamMemberAction {
  type: ActionTypes.TOGGLE_TEAM_MEMBER;
  payload: { memberId: string };
}

// export interface IRemoveTeamMemberAction {
//   type: ActionTypes.REMOVE_TEAM_MEMBER;
//   payload: { memberId: string };
// }

// export interface IAssignTeamAction {
//   type: ActionTypes.ASSIGN_TEAM;
//   payload: { ITeamMember };
// }
