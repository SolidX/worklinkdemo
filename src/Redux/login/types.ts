import { IWorker } from '../../data';
// user Object
export interface IUserLogin {
  id: string;
  username: string;
  password: string;
}

// Constants
export enum ActionTypes {
  LOG_IN = '@@login/LOG_IN',
  // SET_LOGGED_IN_STATE = '@@login/SET_LOGGED_IN_STATE',
  SET_SAVED_USER_STATE = '@@login/SET_SAVED_USER_STATE'
}

export interface ILogin {
  type: ActionTypes.LOG_IN;
  loggedIn: boolean;
  user: IWorker[] | string | null;
}

export interface ISaveUserInAsync {
  type: ActionTypes.SET_SAVED_USER_STATE;
  userSaved: boolean;
}
