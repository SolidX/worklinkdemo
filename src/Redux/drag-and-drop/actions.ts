import {
  ActionTypes,
  IToggleTeamMemberAction
  // IRemoveTeamMemberAction
} from './types';

export const toggleTeamMember = (memberId: string): IToggleTeamMemberAction => {
  return {
    type: ActionTypes.TOGGLE_TEAM_MEMBER,
    payload: { memberId }
  };
};

// export const removeTeamMember = (memberId: string): IRemoveTeamMemberAction => {
//   return {
//     type: ActionTypes.REMOVE_TEAM_MEMBER,
//     payload: { memberId }
//   };
// };

export type Action = IToggleTeamMemberAction;
