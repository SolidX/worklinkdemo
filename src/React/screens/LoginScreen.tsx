import React, { Component } from 'react';
import {
  View,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  TouchableWithoutFeedback,
  Keyboard
} from 'react-native';
import { NavigationScreenProp } from 'react-navigation';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { IState } from '../../../src/Redux';
import { IUserLogin } from '../../../src/Redux/login/types';
import { saveUserInAsync, login } from '../../Redux/login/actions';
import LoginButton from '../components/Buttons/LoginButton';

export interface ILoginNavProps {
  navigation: NavigationScreenProp<any, any>;
}

export interface ILogInProps {
  userSaved: boolean;
  loggedIn: boolean;
  username: string | null;
  user: IUserLogin;
  error: string;
}

export interface ILoginDispatchProps {
  saveUser(user: IUserLogin): void;
  onLogin(username: string, password: string): void;
}

type Props = ILogInProps & ILoginDispatchProps & ILoginNavProps;

class LoginScreen extends Component<Props> {
  state = {
    username: '',
    password: ''
  };
  componentDidMount() {
    const user = { username: 'Test', password: '1234' };
    this.props.saveUser(user);
  }
  handleLogin = () => {
    const username = this.state.username;
    const password = this.state.password;
    this.props.onLogin(username, password);
  };
  render() {
    const shadowStyle = {
      backgroundColor: '#fff',
      shadowOpacity: 0.1,
      elevation: 4,
      shadowOffset: { width: -1, height: 3 },
      shadowRadius: 5
    };
    return (
      <View style={styles.container}>
        <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
          <View style={[styles.loginCard, shadowStyle]}>
            <Text style={styles.title}>NYU Login</Text>
            <TextInput
              style={styles.inputText}
              placeholder={'Net ID'}
              placeholderTextColor={'#a4a4a4'}
              autoCorrect={false}
              returnKeyType="next"
              onSubmitEditing={() => this.txtPassword.focus()}
              value={this.state.username}
              onChangeText={text => this.setState({ username: text })}
            />
            <TextInput
              style={styles.inputText}
              placeholder={'Password'}
              placeholderTextColor={'#a4a4a4'}
              autoCorrect={false}
              returnKeyType="go"
              secureTextEntry={true}
              ref={t => (this.txtPassword = t)}
              value={this.state.password}
              onChangeText={text => this.setState({ password: text })}
            />
            <LoginButton
              handleLogin={this.handleLogin}
              navigation={this.props.navigation}
              loggedIn={this.props.loggedIn}
            />
          </View>
        </TouchableWithoutFeedback>
      </View>
    );
  }
}

// redux
const mapStateToProps = (state: IState) => state.login;
const mapDispatchToProps = (dispatch: Dispatch): ILoginDispatchProps => ({
  saveUser: (user: IUserLogin) => {
    dispatch(saveUserInAsync(user));
  },
  onLogin: (username: string, password: string) => {
    dispatch(login(username, password));
  }
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LoginScreen);

// styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingBottom: '20%',
    backgroundColor: '#f8f8f8'
  },
  loginCard: {
    flex: 0.3,
    backgroundColor: '#fff',
    width: '50%',
    borderRadius: 6,
    justifyContent: 'space-evenly'
  },
  title: {
    fontSize: 16,
    paddingLeft: '5%'
  },
  inputText: {
    width: '85%',
    height: '20%',
    alignSelf: 'center',
    borderWidth: 1,
    borderStyle: 'solid',
    borderColor: '#a4a4a4',
    borderRadius: 5,
    paddingLeft: '4%'
  },
  loginButton: {
    backgroundColor: '#e61950',
    width: '45%',
    height: '15%',
    marginLeft: '30%',
    borderRadius: 15
  },
  loginText: {
    color: '#fff',
    textAlign: 'center',
    paddingTop: '5%',
    fontWeight: 'bold'
  }
});
