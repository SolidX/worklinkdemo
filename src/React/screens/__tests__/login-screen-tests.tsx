import { shallow, configure } from 'enzyme';
import React from 'react';
import configureStore from 'redux-mock-store';
import {} from 'ts-jest';
import thunk from 'redux-thunk';
import Props from '../LoginScreen';
import { ITeamMember } from '../../../Redux/drag-and-drop/types';
// imported as a connected component!
import LoginScreen from '../LoginScreen';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const middleware = [thunk];
const mockStore = configureStore(middleware);
describe('Login Screen', () => {
  const navigation = {
    navigate: jest.fn(),
    dispatch: jest.fn(),
    state: {},
    goBack: jest.fn(),
    dismiss: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    isFocused: jest.fn()
  };
  let wrapper: any, store;

  beforeEach(() => {
    const initialState = {
      userSaved: false,
      loggedIn: false,
      username: '',
      user: false,
      error: ''
    };

    store = mockStore(initialState);
    // Shallow render the container passing in the mock store
    wrapper = shallow(
      <LoginScreen
        navigation={navigation}
        // teamMembers={initialState.teamMembers}
        // error={initialState.error}
      />,
      {
        context: { store }
      }
    );
  });
  it('renders as expected', () => {
    const component = wrapper.dive();
    expect(component).toMatchSnapshot();
  });
});
