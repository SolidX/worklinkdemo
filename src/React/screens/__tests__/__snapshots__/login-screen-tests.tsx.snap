// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginScreen
    navigation={
      Object {
        "addListener": [MockFunction],
        "closeDrawer": [MockFunction],
        "dismiss": [MockFunction],
        "dispatch": [MockFunction],
        "getParam": [MockFunction],
        "goBack": [MockFunction],
        "isFocused": [MockFunction],
        "navigate": [MockFunction],
        "openDrawer": [MockFunction],
        "pop": [MockFunction],
        "popToTop": [MockFunction],
        "push": [MockFunction],
        "replace": [MockFunction],
        "setParams": [MockFunction],
        "state": Object {},
        "toggleDrawer": [MockFunction],
      }
    }
    onLogin={[Function]}
    saveUser={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TouchableWithoutFeedback
        onPress={[Function]}
      >
        <Component
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 6,
                "flex": 0.3,
                "justifyContent": "space-evenly",
                "width": "50%",
              },
              Object {
                "backgroundColor": "#fff",
                "elevation": 4,
                "shadowOffset": Object {
                  "height": 3,
                  "width": -1,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 5,
              },
            ]
          }
        >
          <Component
            style={
              Object {
                "fontSize": 16,
                "paddingLeft": "5%",
              }
            }
          >
            NYU Login
          </Component>
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Net ID"
            placeholderTextColor="#a4a4a4"
            returnKeyType="next"
            style={
              Object {
                "alignSelf": "center",
                "borderColor": "#a4a4a4",
                "borderRadius": 5,
                "borderStyle": "solid",
                "borderWidth": 1,
                "height": "20%",
                "paddingLeft": "4%",
                "width": "85%",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="Password"
            placeholderTextColor="#a4a4a4"
            returnKeyType="go"
            secureTextEntry={true}
            style={
              Object {
                "alignSelf": "center",
                "borderColor": "#a4a4a4",
                "borderRadius": 5,
                "borderStyle": "solid",
                "borderWidth": 1,
                "height": "20%",
                "paddingLeft": "4%",
                "width": "85%",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <LoginButton
            handleLogin={[Function]}
            navigation={
              Object {
                "addListener": [MockFunction],
                "closeDrawer": [MockFunction],
                "dismiss": [MockFunction],
                "dispatch": [MockFunction],
                "getParam": [MockFunction],
                "goBack": [MockFunction],
                "isFocused": [MockFunction],
                "navigate": [MockFunction],
                "openDrawer": [MockFunction],
                "pop": [MockFunction],
                "popToTop": [MockFunction],
                "push": [MockFunction],
                "replace": [MockFunction],
                "setParams": [MockFunction],
                "state": Object {},
                "toggleDrawer": [MockFunction],
              }
            }
          />
        </Component>
      </TouchableWithoutFeedback>,
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#f8f8f8",
        "flex": 1,
        "justifyContent": "center",
        "paddingBottom": "20%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 6,
                "flex": 0.3,
                "justifyContent": "space-evenly",
                "width": "50%",
              },
              Object {
                "backgroundColor": "#fff",
                "elevation": 4,
                "shadowOffset": Object {
                  "height": 3,
                  "width": -1,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 5,
              },
            ]
          }
        >
          <Component
            style={
              Object {
                "fontSize": 16,
                "paddingLeft": "5%",
              }
            }
          >
            NYU Login
          </Component>
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Net ID"
            placeholderTextColor="#a4a4a4"
            returnKeyType="next"
            style={
              Object {
                "alignSelf": "center",
                "borderColor": "#a4a4a4",
                "borderRadius": 5,
                "borderStyle": "solid",
                "borderWidth": 1,
                "height": "20%",
                "paddingLeft": "4%",
                "width": "85%",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="Password"
            placeholderTextColor="#a4a4a4"
            returnKeyType="go"
            secureTextEntry={true}
            style={
              Object {
                "alignSelf": "center",
                "borderColor": "#a4a4a4",
                "borderRadius": 5,
                "borderStyle": "solid",
                "borderWidth": 1,
                "height": "20%",
                "paddingLeft": "4%",
                "width": "85%",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <LoginButton
            handleLogin={[Function]}
            navigation={
              Object {
                "addListener": [MockFunction],
                "closeDrawer": [MockFunction],
                "dismiss": [MockFunction],
                "dispatch": [MockFunction],
                "getParam": [MockFunction],
                "goBack": [MockFunction],
                "isFocused": [MockFunction],
                "navigate": [MockFunction],
                "openDrawer": [MockFunction],
                "pop": [MockFunction],
                "popToTop": [MockFunction],
                "push": [MockFunction],
                "replace": [MockFunction],
                "setParams": [MockFunction],
                "state": Object {},
                "toggleDrawer": [MockFunction],
              }
            }
          />
        </Component>,
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Component
              style={
                Object {
                  "fontSize": 16,
                  "paddingLeft": "5%",
                }
              }
            >
              NYU Login
            </Component>,
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Net ID"
              placeholderTextColor="#a4a4a4"
              returnKeyType="next"
              style={
                Object {
                  "alignSelf": "center",
                  "borderColor": "#a4a4a4",
                  "borderRadius": 5,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": "20%",
                  "paddingLeft": "4%",
                  "width": "85%",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />,
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="Password"
              placeholderTextColor="#a4a4a4"
              returnKeyType="go"
              secureTextEntry={true}
              style={
                Object {
                  "alignSelf": "center",
                  "borderColor": "#a4a4a4",
                  "borderRadius": 5,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": "20%",
                  "paddingLeft": "4%",
                  "width": "85%",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />,
            <LoginButton
              handleLogin={[Function]}
              navigation={
                Object {
                  "addListener": [MockFunction],
                  "closeDrawer": [MockFunction],
                  "dismiss": [MockFunction],
                  "dispatch": [MockFunction],
                  "getParam": [MockFunction],
                  "goBack": [MockFunction],
                  "isFocused": [MockFunction],
                  "navigate": [MockFunction],
                  "openDrawer": [MockFunction],
                  "pop": [MockFunction],
                  "popToTop": [MockFunction],
                  "push": [MockFunction],
                  "replace": [MockFunction],
                  "setParams": [MockFunction],
                  "state": Object {},
                  "toggleDrawer": [MockFunction],
                }
              }
            />,
          ],
          "style": Array [
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 6,
              "flex": 0.3,
              "justifyContent": "space-evenly",
              "width": "50%",
            },
            Object {
              "backgroundColor": "#fff",
              "elevation": 4,
              "shadowOffset": Object {
                "height": 3,
                "width": -1,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 5,
            },
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "NYU Login",
              "style": Object {
                "fontSize": 16,
                "paddingLeft": "5%",
              },
            },
            "ref": null,
            "rendered": "NYU Login",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "autoCorrect": false,
              "onChangeText": [Function],
              "onSubmitEditing": [Function],
              "placeholder": "Net ID",
              "placeholderTextColor": "#a4a4a4",
              "returnKeyType": "next",
              "style": Object {
                "alignSelf": "center",
                "borderColor": "#a4a4a4",
                "borderRadius": 5,
                "borderStyle": "solid",
                "borderWidth": 1,
                "height": "20%",
                "paddingLeft": "4%",
                "width": "85%",
              },
              "underlineColorAndroid": "transparent",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "autoCorrect": false,
              "onChangeText": [Function],
              "placeholder": "Password",
              "placeholderTextColor": "#a4a4a4",
              "returnKeyType": "go",
              "secureTextEntry": true,
              "style": Object {
                "alignSelf": "center",
                "borderColor": "#a4a4a4",
                "borderRadius": 5,
                "borderStyle": "solid",
                "borderWidth": 1,
                "height": "20%",
                "paddingLeft": "4%",
                "width": "85%",
              },
              "underlineColorAndroid": "transparent",
              "value": "",
            },
            "ref": Object {
              "current": null,
            },
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleLogin": [Function],
              "loggedIn": undefined,
              "navigation": Object {
                "addListener": [MockFunction],
                "closeDrawer": [MockFunction],
                "dismiss": [MockFunction],
                "dispatch": [MockFunction],
                "getParam": [MockFunction],
                "goBack": [MockFunction],
                "isFocused": [MockFunction],
                "navigate": [MockFunction],
                "openDrawer": [MockFunction],
                "pop": [MockFunction],
                "popToTop": [MockFunction],
                "push": [MockFunction],
                "replace": [MockFunction],
                "setParams": [MockFunction],
                "state": Object {},
                "toggleDrawer": [MockFunction],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <Component
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "borderRadius": 6,
                  "flex": 0.3,
                  "justifyContent": "space-evenly",
                  "width": "50%",
                },
                Object {
                  "backgroundColor": "#fff",
                  "elevation": 4,
                  "shadowOffset": Object {
                    "height": 3,
                    "width": -1,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 5,
                },
              ]
            }
          >
            <Component
              style={
                Object {
                  "fontSize": 16,
                  "paddingLeft": "5%",
                }
              }
            >
              NYU Login
            </Component>
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Net ID"
              placeholderTextColor="#a4a4a4"
              returnKeyType="next"
              style={
                Object {
                  "alignSelf": "center",
                  "borderColor": "#a4a4a4",
                  "borderRadius": 5,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": "20%",
                  "paddingLeft": "4%",
                  "width": "85%",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="Password"
              placeholderTextColor="#a4a4a4"
              returnKeyType="go"
              secureTextEntry={true}
              style={
                Object {
                  "alignSelf": "center",
                  "borderColor": "#a4a4a4",
                  "borderRadius": 5,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": "20%",
                  "paddingLeft": "4%",
                  "width": "85%",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <LoginButton
              handleLogin={[Function]}
              navigation={
                Object {
                  "addListener": [MockFunction],
                  "closeDrawer": [MockFunction],
                  "dismiss": [MockFunction],
                  "dispatch": [MockFunction],
                  "getParam": [MockFunction],
                  "goBack": [MockFunction],
                  "isFocused": [MockFunction],
                  "navigate": [MockFunction],
                  "openDrawer": [MockFunction],
                  "pop": [MockFunction],
                  "popToTop": [MockFunction],
                  "push": [MockFunction],
                  "replace": [MockFunction],
                  "setParams": [MockFunction],
                  "state": Object {},
                  "toggleDrawer": [MockFunction],
                }
              }
            />
          </Component>
        </TouchableWithoutFeedback>,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#f8f8f8",
          "flex": 1,
          "justifyContent": "center",
          "paddingBottom": "20%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Component
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "borderRadius": 6,
                  "flex": 0.3,
                  "justifyContent": "space-evenly",
                  "width": "50%",
                },
                Object {
                  "backgroundColor": "#fff",
                  "elevation": 4,
                  "shadowOffset": Object {
                    "height": 3,
                    "width": -1,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 5,
                },
              ]
            }
          >
            <Component
              style={
                Object {
                  "fontSize": 16,
                  "paddingLeft": "5%",
                }
              }
            >
              NYU Login
            </Component>
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Net ID"
              placeholderTextColor="#a4a4a4"
              returnKeyType="next"
              style={
                Object {
                  "alignSelf": "center",
                  "borderColor": "#a4a4a4",
                  "borderRadius": 5,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": "20%",
                  "paddingLeft": "4%",
                  "width": "85%",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="Password"
              placeholderTextColor="#a4a4a4"
              returnKeyType="go"
              secureTextEntry={true}
              style={
                Object {
                  "alignSelf": "center",
                  "borderColor": "#a4a4a4",
                  "borderRadius": 5,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": "20%",
                  "paddingLeft": "4%",
                  "width": "85%",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <LoginButton
              handleLogin={[Function]}
              navigation={
                Object {
                  "addListener": [MockFunction],
                  "closeDrawer": [MockFunction],
                  "dismiss": [MockFunction],
                  "dispatch": [MockFunction],
                  "getParam": [MockFunction],
                  "goBack": [MockFunction],
                  "isFocused": [MockFunction],
                  "navigate": [MockFunction],
                  "openDrawer": [MockFunction],
                  "pop": [MockFunction],
                  "popToTop": [MockFunction],
                  "push": [MockFunction],
                  "replace": [MockFunction],
                  "setParams": [MockFunction],
                  "state": Object {},
                  "toggleDrawer": [MockFunction],
                }
              }
            />
          </Component>,
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Component
                style={
                  Object {
                    "fontSize": 16,
                    "paddingLeft": "5%",
                  }
                }
              >
                NYU Login
              </Component>,
              <TextInput
                allowFontScaling={true}
                autoCorrect={false}
                onChangeText={[Function]}
                onSubmitEditing={[Function]}
                placeholder="Net ID"
                placeholderTextColor="#a4a4a4"
                returnKeyType="next"
                style={
                  Object {
                    "alignSelf": "center",
                    "borderColor": "#a4a4a4",
                    "borderRadius": 5,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "height": "20%",
                    "paddingLeft": "4%",
                    "width": "85%",
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />,
              <TextInput
                allowFontScaling={true}
                autoCorrect={false}
                onChangeText={[Function]}
                placeholder="Password"
                placeholderTextColor="#a4a4a4"
                returnKeyType="go"
                secureTextEntry={true}
                style={
                  Object {
                    "alignSelf": "center",
                    "borderColor": "#a4a4a4",
                    "borderRadius": 5,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "height": "20%",
                    "paddingLeft": "4%",
                    "width": "85%",
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />,
              <LoginButton
                handleLogin={[Function]}
                navigation={
                  Object {
                    "addListener": [MockFunction],
                    "closeDrawer": [MockFunction],
                    "dismiss": [MockFunction],
                    "dispatch": [MockFunction],
                    "getParam": [MockFunction],
                    "goBack": [MockFunction],
                    "isFocused": [MockFunction],
                    "navigate": [MockFunction],
                    "openDrawer": [MockFunction],
                    "pop": [MockFunction],
                    "popToTop": [MockFunction],
                    "push": [MockFunction],
                    "replace": [MockFunction],
                    "setParams": [MockFunction],
                    "state": Object {},
                    "toggleDrawer": [MockFunction],
                  }
                }
              />,
            ],
            "style": Array [
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 6,
                "flex": 0.3,
                "justifyContent": "space-evenly",
                "width": "50%",
              },
              Object {
                "backgroundColor": "#fff",
                "elevation": 4,
                "shadowOffset": Object {
                  "height": 3,
                  "width": -1,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 5,
              },
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "NYU Login",
                "style": Object {
                  "fontSize": 16,
                  "paddingLeft": "5%",
                },
              },
              "ref": null,
              "rendered": "NYU Login",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allowFontScaling": true,
                "autoCorrect": false,
                "onChangeText": [Function],
                "onSubmitEditing": [Function],
                "placeholder": "Net ID",
                "placeholderTextColor": "#a4a4a4",
                "returnKeyType": "next",
                "style": Object {
                  "alignSelf": "center",
                  "borderColor": "#a4a4a4",
                  "borderRadius": 5,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": "20%",
                  "paddingLeft": "4%",
                  "width": "85%",
                },
                "underlineColorAndroid": "transparent",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allowFontScaling": true,
                "autoCorrect": false,
                "onChangeText": [Function],
                "placeholder": "Password",
                "placeholderTextColor": "#a4a4a4",
                "returnKeyType": "go",
                "secureTextEntry": true,
                "style": Object {
                  "alignSelf": "center",
                  "borderColor": "#a4a4a4",
                  "borderRadius": 5,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": "20%",
                  "paddingLeft": "4%",
                  "width": "85%",
                },
                "underlineColorAndroid": "transparent",
                "value": "",
              },
              "ref": Object {
                "current": null,
              },
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleLogin": [Function],
                "loggedIn": undefined,
                "navigation": Object {
                  "addListener": [MockFunction],
                  "closeDrawer": [MockFunction],
                  "dismiss": [MockFunction],
                  "dispatch": [MockFunction],
                  "getParam": [MockFunction],
                  "goBack": [MockFunction],
                  "isFocused": [MockFunction],
                  "navigate": [MockFunction],
                  "openDrawer": [MockFunction],
                  "pop": [MockFunction],
                  "popToTop": [MockFunction],
                  "push": [MockFunction],
                  "replace": [MockFunction],
                  "setParams": [MockFunction],
                  "state": Object {},
                  "toggleDrawer": [MockFunction],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "hydrateIn": undefined,
  },
}
`;
